### PR TITLE
Torch binary features

### PR DESCRIPTION
**Pull request by skanjila**
# Code Pull Requests

Please provide the following:

- simple comment

### CODE REVIEW

File: ludwig/modules/loss_modules.py
Comment (w4nderlust) [inline ludwig/modules/loss_modules.py:77]: this is likely wrong. The type and the dict indexing fon't match, and targets and predictions seems flipped.
I would rename the erguments to be predictions and targets, and predictions has logits, not targets.

File: ludwig/modules/loss_modules.py
Comment (w4nderlust) [inline ludwig/modules/loss_modules.py:81]: this hsouldn't be here. Here we are just computing the loss, not computing the gradients

File: ludwig/modules/loss_modules.py
Comment (skanjila) [reply ludwig/modules/loss_modules.py]: Fixed

File: ludwig/modules/loss_modules.py
Comment (skanjila) [reply ludwig/modules/loss_modules.py]: Removed

File: ludwig/modules/loss_modules.py
Comment (w4nderlust) [inline ludwig/modules/loss_modules.py:79]: we can remove

File: ludwig/modules/loss_modules.py
Comment (w4nderlust) [inline ludwig/modules/loss_modules.py:68]: Not really sure why we are creating an IntTensor but giving it type float32

File: ludwig/modules/loss_modules.py
Comment (skanjila) [reply ludwig/modules/loss_modules.py]: removed

File: ludwig/modules/loss_modules.py
Comment (skanjila) [reply ludwig/modules/loss_modules.py]: updated

File: ludwig/encoders/binary_encoders.py
Comment (ShreyaR) [inline ludwig/encoders/binary_encoders.py:51]: Could you remove the `training` and `mask` arguments? I believe we aren't using them anymore in torch.

File: ludwig/encoders/binary_encoders.py
Comment (ShreyaR) [inline ludwig/encoders/binary_encoders.py:57]: Could you also add `input_shape` and `output_shape` properties for the class? Example here for [category encoders](https://github.com/ludwig-ai/ludwig/blob/ee17666addc269cb022272c55edba58b6dbaff97/ludwig/encoders/category_encoders.py#L85).

File: ludwig/features/binary_feature.py
Comment (ShreyaR) [inline ludwig/features/binary_feature.py:19]: Could you remove the commented out code from this file? Also at lines 20, 30, 136, etc.

File: ludwig/features/binary_feature.py
Comment (ShreyaR) [inline ludwig/features/binary_feature.py:240]: Could you also add an `input_shape` property for this feature?

File: tests/integration_tests/test_binary_feature.py
Comment (ShreyaR) [inline tests/integration_tests/test_binary_feature.py:1]: Thanks for adding this test!

Could you move this to `tests/ludwig/features`?

**Review by ShreyaR — COMMENTED** [review]
Thanks for porting this feature @skanjila! I left some minor comments..

File: ludwig/encoders/binary_encoders.py
Comment (skanjila) [reply ludwig/encoders/binary_encoders.py]: removed

File: ludwig/encoders/binary_encoders.py
Comment (skanjila) [reply ludwig/encoders/binary_encoders.py:56]: added

File: ludwig/features/binary_feature.py
Comment (skanjila) [reply ludwig/features/binary_feature.py:19]: removed all commented out code

File: ludwig/features/binary_feature.py
Comment (skanjila) [reply ludwig/features/binary_feature.py:221]: added

File: tests/integration_tests/test_binary_feature.py
Comment (skanjila) [reply tests/integration_tests/test_binary_feature.py]: moved, although Jim's tests that follow this pattern live in the integration tests directory

Comment (ShreyaR) [reply tests/integration_tests/test_binary_feature.py]: I see. I can add a fix for that during the testing phase.

**Review by ShreyaR — APPROVED** [review]
lgtm! Feel free to merge the PR once @w4nderlust approves it as well.

File: ludwig/modules/loss_modules.py
Comment (w4nderlust) [inline ludwig/modules/loss_modules.py:70]: change np.log with torch.log

**Pull request merged**
